pkg/datastore/datastore.go

```diff
@@ -817,10 +817,17 @@ type Revision interface {
 
 	// LessThan returns whether the receiver is probably less than the right hand side.
 	LessThan(Revision) bool
+
+	// ByteSortable returns true if the string representation of the Revision is byte sortable, false otherwise.
+	ByteSortable() bool
 }
 
 type nilRevision struct{}
 
+func (nilRevision) ByteSortable() bool {
+	return true
```

Comment:
yeah, I though the `nil` value was a string, and is sortable 😅 I wasn't expecting use of it that way.
I'm ok with your proposal 👍🏻

Reply:
Actually, all parsing functions rely on `string`, so there would be a mismatch between returning `[]byte`, and parsing it, which has to be converted to `string`, and while that would work, it may not be the case if we ever choose to represent the bytes differently.

Reply:
Maybe just return `false` for nilRevision then?